### PR TITLE
FE - Add an Event for a Swim Meet

### PR DESCRIPTION
**Pull request by viriponce**
This PR addresses issue #93.

### Implementation

It implements a feature that allows users to add a new event for an specific Swim Meet when clicking the _"+EVENT"_ button from the Meet Event Display or from the success alert box shown when a new Swim Meet is created.

1. New components in the Event folder:
- AddEventForm
This component implements a form to submit the information for a new swim meet event. It displays a select to chose a Group and another select to select an Even Type. It receives the options for these selects as props.
It includes a create and cancel button. The methods to handle these actions are received as props.

- AddEvent
It's the component that implements the logic to add a new event to an specific swim meet.
It's the parent component to `AddEventForm`.

It implements the logic for the submission method called when clicking the Create button. In which case, the `createEvent` request is sent with the user selected data. It implements also error handling.

In case of an error, it is displayed in an Alert Box. In a similar way, a success message is displayed when the new event is created.

2. Modifications to previous components
- MeetEventDisplay:
The `handleAddNew` function is updated to navigate to the `AddEvent` component. It uses the navigate function to route to `/add-event/${meetId}`, passing the current Swim Meet's data as state.

- AddSwimMeet:
The `handleAddEvents` function is updated to navigate to the `AddEvent` component. It uses the navigate function to route to `/add-event/${lastSwimMeetId}`, passing the just created swim meet id as parameter. This function is called when clicking the +EVENTS button displayed along with the successful create swim meet message.

- MySelect
The styling property `minWidth` is now included with a value of 150.

3. App.jsx

Added the protected route with path="/add-event/:meetId" to navigate to `AddEvent`.

4. App.css
The code format is updated.

5. SmmApp

- `getGroups` method:
This method fetches the possible Groups.  
 
- `getEventTypes` method
This method fetches the possible Event Types.  

- `createEvent` method
This method post  the data for a new event for a specific Swim Meet id.  
  - Parameters:
    - `meetId`: The ID of the Swim Meet for which the event is being created.

### UI Preview

Event creation form
Screenshot 2024-09-12 at 4 53 16 PM

Event creation form with selected values for a new event and after  successfully submission:
Screenshot 2024-09-12 at 6 17 35 AM

Event creation form with selected values for a new event and after  unsuccessful submission, displaying an error:
Screenshot 2024-09-12 at 6 17 00 AM

Error message when unable to load the add event form:
Screenshot 2024-09-12 at 6 16 13 AM

### CODE REVIEW

**Review by MarisiaS — CHANGES_REQUESTED** [review]
I think the AddEvent should include a button to navigate directly to the list of events for the swim meet.

When creating a new swim meet, we didn't add a "back to swim meets" button since the navigation already provides that option.
However, in the AddEvent, having a button to go directly to the list of events for the swim meet would improve the user experience. Otherwise, users would need to go back to the swim meets list, search for the swim meet they're working on, and then click "Go to event."  This extra step is not very user-friendly.

**Comment by MarisiaS** [comment]
I just realized that the cancel button currently sends the user to the list of events.
It might be better to change the label of that button to something like "Back to Events" 
to make its function clearer and improve the user experience. What do you think?

File: frontend/src/SwimMeet/AddSwimMeet.jsx [hunk 1]
```diff
@@ -84,8 +84,7 @@ const AddSwimMeet = () => {
   };
 
   const handleAddEvents = () => {
-    //Change it to add events for the swim meet generated
-    navigate(`/events`);
+    navigate(`/add-event/${lastSwimMeetId}`);
```
[the remark this answers is not in the document]
Comment (viriponce) [reply frontend/src/SwimMeet/AddSwimMeet.jsx]: The response data i.e. swim meet information is now set to the state lastSwimMeetData and pass it when navigating to AddEvent.

**Comment by viriponce** [comment]
> I just realized that the cancel button currently sends the user to the list of events. It might be better to change the label of that button to something like "Back to Events" to make its function clearer and improve the user experience. What do you think?


As discussed, the button is renamed to "Back to Events" and the layout is updated.

**Review by MarisiaS — APPROVED** [review]
LGTM!

Thanks Viri!

**Pull request merged**